src/lcd/mod.rs

```diff
@@ -297,6 +297,13 @@ impl<'a, T: Framebuffer> TextWriter<'a, T> {
             self.layer.clear();
         }
     }
+    fn carriage_return(&mut self) {
+        self.x_pos = 0;
```

Comment:
Please call `carriage_return` from `newline` and remove all the other code except `self.y_pos += 8` from `newline`. Otherwise we have code duplication

Reply:
done